solr/core/src/java/org/apache/solr/filestore/DistribPackageStore.java

```diff
@@ -572,7 +582,16 @@ private void ensurePackageStoreDir(Path solrHome) {
   }
 
   public static Path getPackageStoreDirPath(Path solrHome) {
-    return solrHome.resolve(PackageStoreAPI.PACKAGESTORE_DIRECTORY);
+    var path = solrHome.resolve(PackageStoreAPI.PACKAGESTORE_DIRECTORY);
+    if (!Files.exists(path)) {
+      try {
+        Files.createDirectories(path);
+        log.info("Created filestore folder {}", path);
+      } catch (IOException e) {
+        throw new SolrException(SERVER_ERROR, "Faild creating 'filestore' folder in SOLR_HOME. Check permissions");
```

Comment:
Failed

Reply:
Should we set the Ioexception as cause?

Reply:
There might be a permissions issue; this isn't a problem per-se... can't we just return null and ultimately just disable the package store?  Log something too... I'd just do INFO level.

Reply:
Per definition, SOLR_HOME should be writable, but if not, we should fast instead of risking a NPE somewhere else. This code is not executed until you start interacting with package store.